base/dates/arithmetic.jl

```diff
@@ -24,74 +24,74 @@ end
+    mm = monthwrap(m, value(z)); ld = daysinmonth(ny, mm)
+    return DateTime(ny, mm, d <= ld ? d : ld, hour(dt), minute(dt), second(dt), millisecond(dt))
 end
-function (+)(dt::Date,z::Month)
+function (+)(dt::Date, z::Month)
     y,m,d = yearmonthday(dt)
-    ny = yearwrap(y,m,value(z))
-    mm = monthwrap(m,value(z)); ld = daysinmonth(ny,mm)
-    return Date(ny,mm,d <= ld ? d : ld)
+    ny = yearwrap(y, m, value(z))
+    mm = monthwrap(m, value(z)); ld = daysinmonth(ny, mm)
+    return Date(ny, mm, d <= ld ? d : ld)
 end
-function (-)(dt::DateTime,z::Month)
+function (-)(dt::DateTime, z::Month)
     y,m,d = yearmonthday(dt)
-    ny = yearwrap(y,m,-value(z))
-    mm = monthwrap(m,-value(z)); ld = daysinmonth(ny,mm)
-    return DateTime(ny,mm,d <= ld ? d : ld,hour(dt),minute(dt),second(dt),millisecond(dt))
+    ny = yearwrap(y, m, -value(z))
+    mm = monthwrap(m, -value(z)); ld = daysinmonth(ny, mm)
+    return DateTime(ny, mm, d <= ld ? d : ld, hour(dt), minute(dt), second(dt), millisecond(dt))
 end
-function (-)(dt::Date,z::Month)
+function (-)(dt::Date, z::Month)
     y,m,d = yearmonthday(dt)
-    ny = yearwrap(y,m,-value(z))
-    mm = monthwrap(m,-value(z)); ld = daysinmonth(ny,mm)
-    return Date(ny,mm,d <= ld ? d : ld)
+    ny = yearwrap(y, m, -value(z))
+    mm = monthwrap(m, -value(z)); ld = daysinmonth(ny, mm)
+    return Date(ny, mm, d <= ld ? d : ld)
 end
-(+)(x::Date, y::Week) = return Date(UTD(value(x) + 7*value(y)))
-(-)(x::Date, y::Week) = return Date(UTD(value(x) - 7*value(y)))
+(+)(x::Date, y::Week) = return Date(UTD(value(x) + 7 * value(y)))
+(-)(x::Date, y::Week) = return Date(UTD(value(x) - 7 * value(y)))
 (+)(x::Date, y::Day)  = return Date(UTD(value(x) + value(y)))
 (-)(x::Date, y::Day)  = return Date(UTD(value(x) - value(y)))
 (+)(x::DateTime, y::Period) = return DateTime(UTM(value(x) + toms(y)))
 (-)(x::DateTime, y::Period) = return DateTime(UTM(value(x) - toms(y)))
 (+)(x::Time, y::TimePeriod) = return Time(Nanosecond(value(x) + tons(y)))
 (-)(x::Time, y::TimePeriod) = return Time(Nanosecond(value(x) - tons(y)))
 (+)(y::Period, x::TimeType) = x + y
-(-)(y::Period, x::TimeType) = x - y
 
 for op in (:+, :-)
     @eval begin
-        # GeneralPeriod, AbstractArray{TimeType}
-        ($op){T<:TimeType}(x::AbstractArray{T}, y::GeneralPeriod) = broadcast($op,x,y)
-        ($op){T<:TimeType}(y::GeneralPeriod, x::AbstractArray{T}) = broadcast($op,x,y)
-
-        # TimeType, StridedArray{GeneralPeriod}
-        ($op){T<:TimeType,P<:GeneralPeriod}(x::StridedArray{P}, y::T) = broadcast($op,x,y)
-        ($op){P<:GeneralPeriod}(y::TimeType, x::StridedArray{P}) = broadcast($op,x,y)
+        ($op){T<:TimeType}(x::AbstractArray{T}, y::GeneralPeriod) = broadcast($op, x, y)
+    end
+    if op == :+
+        @eval begin
+            ($op){T<:TimeType}(y::GeneralPeriod, x::AbstractArray{T}) = broadcast($op, x, y)
+            ($op){T<:TimeType,P<:GeneralPeriod}(x::StridedArray{P}, y::T) = broadcast($op, x, y)
```

Comment:
since the argument order of this one is `x, y` on both lhs and rhs I think this one is fine to have defined for subtraction